core/src/test/scala-3/sttp/tapir/SchemaMacroScala3Test.scala

```diff
@@ -88,6 +88,28 @@ class SchemaMacroScala3Test extends AnyFlatSpec with Matchers:
     coproduct.subtypeSchema(true).map(_.schema.schemaType) shouldBe Some(SchemaType.SBoolean())
   }
 
+  it should "derive schema for a string-based union type" in {
+    // when
+    val s: Schema["a" | "b"] = Schema.derivedStringBasedUnionEnumeration
+
+    // then
+    s.name.map(_.show) shouldBe Some("a_or_b")
+
+    s.schemaType should matchPattern { case SchemaType.SString() => }
+    s.validator should matchPattern { case Validator.Enumeration(List("a", "b"), _, _) => }
+  }
+
+  it should "derive schema for a const as a string-based union type" in {
+    // when
+    val s: Schema["a"] = Schema.constStringToEnum
```

Comment:
isn't this a special case of the union-type enumeration? (with a single element)

Reply:
What do you mean? I think a const value should be treated as a single-value enum. The nice thing with string-based enums is that they can be merge, e.g.
```
type AorB = "A" | "B"
type C = "C"
type AorBorC = AorB | C
```
Everything becomes structural, saves a lot of enum headaches

Reply:
yes, I agree, I'm wondering why it's a different derivation method than with the two-values case

also, would be great to add a test using the types as in your comment :)

Reply:
Turns out it could be simplified 🙃